rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -2516,15 +2516,10 @@ private <T extends TypeTree & Expression> T typeTree(@Nullable ClassNode classNo
                     && classNode.getUnresolvedSuperClass().isUsingGenerics()
                     && !classNode.getUnresolvedSuperClass().isGenericsPlaceHolder()
                     && classNode.getGenericsTypes() == null;
-            if (isAnonymousClassWithGenericSuper) {
+            if (isAnonymousClassWithGenericSuper || (classNode.isUsingGenerics() && !classNode.isGenericsPlaceHolder())) {
```

Comment:
ℹ️ This change is just a small refactor. The implementation was completely the same, except for the argument of the `visitTypeParameterizations` call. I figured it's way easier to understand if we collapse these two branches into one.